libvirt.go

```diff
@@ -146,22 +147,36 @@ func (d *Driver) GetURL() (string, error) {
 	return "", nil
 }
 
-func (d *Driver) getConn() (*libvirt.Connect, error) {
-	if d.conn == nil {
-		conn, err := libvirt.NewConnect(connectionString)
+func (d *Driver) getSessionConn() (*libvirt.Connect, error) {
+	log.Debugf("Connecting to session")
+	if d.sessionConn == nil {
+		conn, err := libvirt.NewConnect(sessionConnectionString)
+		if err != nil {
+			log.Errorf("Failed to connect to libvirt: %s", err)
+			return &libvirt.Connect{}, errors.New("Unable to connect to kvm driver")
+		}
+		d.sessionConn = conn
+	}
+	return d.sessionConn, nil
+}
+
+func (d *Driver) getSystemConn() (*libvirt.Connect, error) {
+	log.Debugf("Connecting to system")
+	if d.systemConn == nil {
+		conn, err := libvirt.NewConnectReadOnly(systemConnectionString)
```

Comment:
Only ReadOnly connection, good progress ;) I suspect it's going to break this though:
```
if active, err := network.IsActive(); !active {
}
```
but crc will fail early if the libvirt network is not active when running crc start, so not really useful to have this in the machine driver as well

Reply:
you mean, machine driver can't activate the connection if it's not already active? If so, yeah you're right that `crc` will ensure that for us with it's preflight checks.

Reply:
I mean that this PR switches to using a RO connection, but some code is still there which is going to break on such a connection..

Reply:
where? which code requires write-access still?

Reply:
The code we have been talking about in this thread?
https://github.com/code-ready/machine-driver-libvirt/blob/841bfbf9108d6888c91a2cd83fb062179296725e/libvirt.go#L216

Reply:
Ah! missed that. :)